api/users/permissions.py

```diff
@@ -34,3 +34,16 @@ def has_object_permission(self, request, view, obj):
             return True
         else:
             return obj['self']._id == request_user._id
+
+class ClaimUserPermission(permissions.BasePermission):
+    """ Allows anyone to attempt to claim an unregistered user.
+    Allows no one to attempt to claim a registered user.
+    """
+    def has_permission(self, request, view):
+        claimed_user = view.get_user(check_permissions=False)
+        assert isinstance(claimed_user, OSFUser), 'obj must be a User, got {}'.format(claimed_user)
+        return not claimed_user.is_registered
+
+    def has_object_permission(self, request, view, obj):
+        assert isinstance(obj, OSFUser), 'obj must be a User, got {}'.format(obj)
+        return not obj.is_registered
```

Comment:
Disabled accounts get set is_registered = False so they would get past here. Might that cause problems?

Aside, the "claim user" dialogue shows up for disabled accounts, but breaks when attempted, so there are definitely issues with that workflow that are unrelated to this PR.

Reply:
It seems like disabled accounts shouldn't have `is_registered` set to `False`. A `disabled` `NonNaiveDateTimeField` is probably the more idiomatic way to do that... and we've already got one. The refactor to tear out any use of `is_registered` in that flow is out of scope for this.

I can add a check for `is_claimed` here as well, but that might have other side effects, based on [the TODO above `is_claimed`](https://github.com/CenterForOpenScience/osf.io/blob/65ded1a/osf/models/user.py#L177-L180).

Reply:
Update: Added a check in the view to `raise ValidationError('Cannot claim disabled account.')`